### PR TITLE
Onboard Azure VM and Arc connected machines to Azure Automation DSC

### PR DESCRIPTION
**Pull request by mgreenegit**

### CODE REVIEW

File: samples/Automation/onboard-to-automation-dsc/azurepolicy.json [hunk 1]
```diff
@@ -0,0 +1,307 @@
+{
+  "properties": {
+    "displayName": "Onboard Azure VM and Arc connected machines to Azure Automation DSC",
+    "mode": "Indexed",
+    "description": "Deploys the DSC extension to onboard nodes to Azure Automation DSC. Does not assign a configuration.",
+    "metadata": {
+      "category": "Automation"
+    },
+    "policyRule": {
+      "mode": "Indexed",
```
Comment (pilor) [inline samples/Automation/onboard-to-automation-dsc/azurepolicy.json:10]: this is duplicated

File: samples/Automation/onboard-to-automation-dsc/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+$definition
+$assignment = New-AzPolicyAssignment -Name <assignmentname> -Scope <scope>  -PolicyDefinition $definition
+$assignment 
+````
+
+## Try with CLI
+
+````cli
+az policy definition create --name 'onboard-to-automation-dsc' --display-name 'Onboard Azure VM and Arc connected machines to Azure Automation DSC' --description 'Deploys the DSC extension to onboard nodes to Azure Automation DSC. Does not assign a configuration.' --rules 'https://raw.githubusercontent.com/Azure/azure-policy/master/samples/Automation/onboard-to-automation-dsc/azurepolicy.rules.json' --params 'https://raw.githubusercontent.com/Azure/azure-policy/master/samples/Automation/onboard-to-automation-dsc/azurepolicy.parameters.json' --mode All
```
Comment (pilor) [inline samples/Automation/onboard-to-automation-dsc/README.md:21]: >All [](start = 554, length = 3)

Indexed

File: samples/Automation/onboard-to-automation-dsc/azurepolicy.json (hunk 1 above)
Comment (mgreenegit) [reply samples/Automation/onboard-to-automation-dsc/azurepolicy.json]: thanks @pilor

File: samples/Automation/onboard-to-automation-dsc/README.md (hunk 2 above)
Comment (mgreenegit) [reply samples/Automation/onboard-to-automation-dsc/README.md]: thanks @pilor

**Comment by mgreenegit** [comment]
@pilor sorry, was out sick a couple days. I have made all requested changes and added a second sample for DSC extension without Azure Automation.

**Comment by mgreenegit** [comment]
I'm not sure how to address the json validation test?

File: samples/Compute/deploy-dsc-extension/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+## Deploy DSC Extension to Azure VM and Arc connected machines
+
+Deploys the DSC extension to and assigns configuration artifact from url location.
+
+## Try on Portal
+
+[![Deploy to Azure](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#blade/Microsoft_Azure_Policy/CreatePolicyDefinitionBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-policy%2Fmaster%2Fsamples%2FAutomation%2Fdeploy-dsc-extension%2Fazurepolicy.json)
```
Comment (pilor) [inline samples/Compute/deploy-dsc-extension/README.md:7]: All URLs should use samples/Compute/ in this file

File: samples/Compute/deploy-dsc-extension/azurepolicy.json [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+              "parameters": {
+                "configurationUrl": {
+                  "type": "String"
+                },
+                "configurationScript": {
+                  "type": "String"
+                },
+                "configurationFunction": {
+                  "type": "String"
+                },
```
Comment (pilor) [inline samples/Compute/deploy-dsc-extension/azurepolicy.json:213]: These should be getting values

File: samples/Compute/deploy-dsc-extension/azurepolicy.rules.json [hunk 5]
```diff
@@ -0,0 +1,286 @@
+
```
Comment (pilor) [inline samples/Compute/deploy-dsc-extension/azurepolicy.rules.json:1]: Missing opening {, that is why the test is failing

File: samples/Compute/deploy-dsc-extension/azurepolicy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      }
+    },
+    "configurationFunction": {
+      "type": "String",
+      "metadata": {
+        "displayName": "Configuration name",
+        "description": "The name of the configuration provided in the script file inside the .zip artifact"
+      }
+    }
+  }
```
Comment (pilor) [inline samples/Compute/deploy-dsc-extension/azurepolicy.parameters.json:23]: indentation

Comment (mgreenegit) [reply samples/Compute/deploy-dsc-extension/azurepolicy.parameters.json]: fixed

File: samples/Compute/deploy-dsc-extension/azurepolicy.rules.json (hunk 5 above)
Comment (mgreenegit) [reply samples/Compute/deploy-dsc-extension/azurepolicy.rules.json]: fixed

File: samples/Compute/deploy-dsc-extension/azurepolicy.json (hunk 4 above)
Comment (mgreenegit) [reply samples/Compute/deploy-dsc-extension/azurepolicy.json]: fixed

File: samples/Compute/deploy-dsc-extension/README.md (hunk 3 above)
Comment (mgreenegit) [reply samples/Compute/deploy-dsc-extension/README.md]: fixed

**Pull request merged**